src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -171,6 +174,48 @@ void VehicleInfoPlugin::DeleteSubscriptions(
     }
   }
 }
+
+application_manager::VehicleInfoSubscriptions
+VehicleInfoPlugin::SelectVehicleDataForUnsubscribe(
+    const app_mngr::ApplicationSharedPtr& application) {
+  auto vehicle_extension =
+      VehicleInfoAppExtension::ExtractVIExtension(*application);
+  auto vehicle_data_for_unsubscribe = vehicle_extension.Subscriptions();
+  const application_manager::ApplicationSet& applications =
+      application_manager_->applications().GetData();
```

Comment:
Would this `DataAccessor` do anything if it goes out of scope immediately, as it does now? I think you need to keep a reference to the `DataAccessor` in this case.